### ISSUE TITLE
Support method references in the cucumber-java8 to cucumber-java migration

### ISSUE THREAD

**Issue by timtebeek** [openrewrite/rewrite-cucumber-jvm#3]
### 🤔 What's the problem you're trying to solve?

When migrating from cucumber-java8 to cucumber-java, method references are not supported.

### ✨ What's your proposed solution?

Support method references in the cucumber-java8 to cucumber-java migration recipes.

### ⛏ Have you considered any alternatives or workarounds?

Replace the method references with lambda's before conversion, through an inverse of [Use method references in lambda](https://docs.openrewrite.org/reference/recipes/java/cleanup/replacelambdawithmethodreference).

### 📚 Any additional context?

Likely more important only if we do indeed [deprecate cucumber-java8](https://github.com/cucumber/cucumber-jvm/issues/2174).

**Comment by timtebeek** [comment]
## Implementation plan

### Where it stands today

Both recipes already detect method references and deliberately bail out:

- `CucumberJava8StepDefinitionToCucumberJava` requires the second argument to be a `J.Lambda` assignable to `StepDefinitionBody`, and otherwise returns `SearchResult.found(m, "TODO Migrate manually")` — [`CucumberJava8StepDefinitionToCucumberJava.java#L87-L91`](../blob/main/src/main/java/org/openrewrite/cucumber/jvm/CucumberJava8StepDefinitionToCucumberJava.java#L87-L91)
- `CucumberJava8HookDefinitionToCucumberJava` rejects any argument that is not a `J.Literal` or `J.Lambda` — [`CucumberJava8HookDefinitionToCucumberJava.java#L79-L82`](../blob/main/src/main/java/org/openrewrite/cucumber/jvm/CucumberJava8HookDefinitionToCucumberJava.java#L79-L82)

There are already tests pinning the current (non-)behaviour, so the work is largely a matter of turning their expected output into real migrations:

- `CucumberJava8ToCucumberJavaTest#replaceMethodReference` — `Given("{int} plus {int}", Integer::sum)`
- `CucumberJava8ToCucumberJavaTest#convertMethodReference` — `Before(this::connect)`

Note that the workaround suggested in the issue description is not currently available: rewrite-static-analysis ships `ReplaceLambdaWithMethodReference` and `UseLambdaForFunctionalInterface` (anonymous class → lambda), but nothing that expands a `J.MemberReference` back into a `J.Lambda`. So this has to be solved here.

### Step 1 — a shared `J.MemberReference` → `J.Lambda` expansion

Rather than teaching both recipes to handle `J.MemberReference` alongside `J.Lambda`, add one package-private visitor (say `ExpandMethodReferenceToLambda`) that runs as a precondition step and rewrites method references **only where they are passed to a matched `io.cucumber.java8` glue method**. Everything downstream — method naming, block wrapping, `RemoveUnneededBlock`, `UnnecessaryThrows` in `CucumberJava8ClassVisitor` — then works unchanged.

Building the lambda:

- Parameter types come from the enclosing invocation's `getMethodType().getParameterTypes()`, taking the functional-interface parameter (`StepDefinitionBody.A2<Integer, Integer>`, `HookBody`, …) and reading `JavaType.Parameterized#getTypeParameters()`. This is more reliable than inspecting the member reference itself, since the reference's own signature can differ (`Integer::sum` returns `int`, whereas `StepDefinitionBody.A2#accept` is `void`).
- Parameter names come from `memberRef.getMethodType().getParameterNames()` when available, falling back to `arg0`…`argN`.
- The body is `<containing>.<name>(arg0, …)`, where `<containing>` is `memberRef.getContaining()`. Constructor references (`Foo::new`) need the `new Foo(...)` form.

That alone makes both failing tests migrate, and it also unblocks method references passed to `DataTableType`/`ParameterType` bodies once #2 lands.

### Step 2 — annotate in place for hooks that reference a local method

Expanding `Before(this::connect)` into a generated wrapper that calls `connect()` compiles, but the nicer output is the one the existing test comment already suggests: annotate the referenced declaration directly.

```java
// before
public HookStepDefinitions() {
    Before(this::connect);
}
private void connect() { }

// after
@Before
public void connect() { }
```

This means a new code path in `CucumberJava8ClassVisitor`, which today only ever appends a templated method. Apply it when:

- the reference is `this::name` or `EnclosingClass::name`, and
- the target declaration resolves to a method in the same class, and
- that declaration is not already carrying a conflicting non-repeatable cucumber annotation (`@Before`/`@After` are not `@Repeatable`, so a method referenced twice with different tag expressions must fall back to step 1's wrapper form).

The tag expression and order arguments carry over unchanged — `Before("@wip", 10, this::connect)` becomes `@Before(order = 10, value = "@wip")` on `connect`, reusing `HookArguments#formatAnnotationArguments`. Visibility is widened to `public`, since cucumber-java requires glue methods to be public.

### Tests

- Update `replaceMethodReference` and `convertMethodReference` to assert real migrations.
- Add coverage for: static references (`Integer::sum`), instance references on another object, `EnclosingClass::instanceMethod`, constructor references, a hook whose target takes a `Scenario`, a method referenced by two different hooks, and a reference to a method declared in a superclass (should stay on the wrapper path).

### Adjacent, out of scope here

The anonymous-inner-class cases in the same test file (`Before(new HookNoArgsBody() { … })`) are still marked `TODO Migrate manually`. Those could be picked up almost for free by running `UseLambdaForFunctionalInterface` as an earlier step in the `CucumberJava8ToJava` composite — worth a separate issue.
